os/include/tinyara/pm/pm.h

```diff
@@ -389,6 +322,27 @@ extern "C" {
 
 void pm_initialize(void);
 
+/****************************************************************************
+ * Name: pm_idle
+ *
+ * Description:
+ *   pm_idle() is the logic that will be executed when there is no other
+ *   ready-to-run task.  This is processor idle time and will continue until
+ *   some interrupt occurs to cause a context switch from the idle task.
+ *
+ *   Processing in this state may be processor-specific. e.g., this is where
+ *   power management operations might be performed.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned value:
+ *   None
+ *
+ ****************************************************************************/
+
+void pm_idle(void);
```

Comment:
What I guess operation of this function from name is put a device to idle.
But actually this function changes a state of pm only. up_idle puts a device to idle.
1. How about change a function name ? i.e. pm_prepare_idle, pm_enter_idle_state, or something else?
2. If you want to keep this name, how about put up_idle into pm_idle?
in os_start.c
```
#ifdef CONFIG_PM
pm_idle();
#else
up_idle();
#endif
```

Reply:
Function of these API's:
pm_idle() - PM is supported
      To make decision about which state to switch to next & then change to new state.
up_idle() - PM is not supported
       To put system to WFI mode

Will change current implementation to:
#ifdef CONFIG_PM
pm_idle();
#else
up_idle();
#endif

As up_idle cannot be a part of pm_idle beacuse it is architecture dependent code & pm_idle is part of generic framework.

Reply:
I don't mean adding s5j_idle inside pm_idle. The up_idle is a common API regardless of architecture. Implementation of up_idle is arch-dependant. That is same as up_putc. Implementation is different by archi but API is a common so that serial driver calls it.
As far as I know, pm is highly dependant with archi and pm idle should put a device to idle.
Without up_idle, I can not know why pm_idle is needed.

Reply:
If you will not put up_idle into pm_idle, you should not modify as you said.
```
Will change current implementation to:
#ifdef CONFIG_PM
pm_idle();
#else
up_idle();
#endif
```

Reply:
As pm_idle is part of generic PM now it:
1. decides next state
2. calls API provided by SoC (up_pm_switch state) to switch to next state

 I will keep implementation as discussed over messenger:
#ifdef CONFIG_PM
pm_idle();
#endif
up_idle();